benches/search_comparison.rs

```diff
@@ -167,111 +165,99 @@ fn search_bench_case<const MAX: usize, T, Coll>(
     setup_fun: impl Fn(Vec<T>) -> Coll,
     search_fun: impl Fn(&Coll, T) -> Option<&T>,
     group: &mut BenchmarkGroup<WallTime>,
-    size: &usize,
+    size: usize,
     duplicates: bool,
 ) where
-    T: TryFrom<usize> + Ord + Clone,
-    <T as TryFrom<usize>>::Error: core::fmt::Debug,
+    T: TryFrom<usize> + Ord + Copy,
 {
-    group.bench_with_input(BenchmarkId::new(name, size), size, |b, &size| {
+    group.bench_with_input(BenchmarkId::new(name, size), &size, |b, &size| {
         // increasing sequence of even numbers, bounded by MAX
         let iter = (0usize..)
             // Generate only even numbers to provide a ~50% hit ratio in the benchmark
             .map(|i| (i * 2) % MAX)
-            .map(|i| T::try_from(i).unwrap());
+            .map(|i| T::try_from(i).ok().unwrap());
```

Comment:
Why this change?

Reply:
This change converts `Error<T, E>` into `Option<T>`. This way there is no need in `<T as TryFrom<usize>>::Error: core::fmt::Debug` trait because of `unwrap()`, and functions signatures are a little bit more concise. From diagnostic point of view unwrapping the `Result` is the same as unwrapping `Option`, there is no any additional information in `Result`. At least in this version of code.